crates/chain-state/src/notifications.rs

```diff
@@ -211,6 +212,86 @@ impl<T: Clone + Sync + Send + 'static> Stream for ForkChoiceStream<T> {
     }
 }
 
+/// A stream that emits canonical chain notifications only when blocks are finalized.
+#[pin_project]
+pub(crate) struct FinalizedNotificationStream<N: NodePrimitives> {
+    #[pin]
+    canon_stream: CanonStateNotificationStream<N>,
+    #[pin]
+    finalized_stream: ForkChoiceStream<SealedHeader<alloy_consensus::Header>>,
```

Comment:
we need to replace the concrete header types with 
```suggestion
    finalized_stream: ForkChoiceStream<SealedHeaderFor<N>>,
```